packages/mapviewer/tests/cypress/tests-e2e/search/search-results.cy.js

```diff
@@ -44,9 +45,49 @@ function checkDescendantOf(selector, message) {
         return $element
     }
 }
+const acceptedDelta = 0.1
+const checkLocation = (expected, result) => {
+    expect(result).to.be.an('Array')
+    expect(result.length).to.eq(2)
+    expect(result[0]).to.approximately(expected[0], acceptedDelta)
+    expect(result[1]).to.approximately(expected[1], acceptedDelta)
+}
+
+function testQueryPositionCrosshairStore({
+    searchQuery,
+    expectedCenter,
+    expectedPinnedLocation,
+    expectedCrosshair = null,
+    expectedCrosshairPosition = null,
+}) {
+    // check the query
+    cy.readStoreValue('state.search.query').should('eq', searchQuery)
+    // check the center of the map
+    cy.readStoreValue('state.position.center').should((positionCenter) =>
+        checkLocation(expectedCenter, positionCenter)
+    )
+    // check the location of pinnedLocation
+    cy.readStoreValue('state.map.pinnedLocation').should((pinnedLocation) =>
+        checkLocation(expectedPinnedLocation, pinnedLocation)
+    )
+    // check the crosshair
+    if (expectedCrosshair !== null) {
+        cy.readStoreValue('state.position.crossHair').should('eq', expectedCrosshair)
+    } else {
+        cy.readStoreValue('state.position.crossHair').should('be.null')
+    }
+    // check the crosshair position
+    if (expectedCrosshairPosition !== null) {
+        cy.readStoreValue('state.position.crossHairPosition').should((crossHairPosition) =>
+            checkLocation(expectedCrosshairPosition, crossHairPosition)
+        )
+    } else {
+        cy.readStoreValue('state.position.crossHairPosition').should('be.null')
+    }
```

Comment:
I was thinking if it would be better to have the null verification happen in the `checkLocation` function, but I think having it here is easier to read about what happens.